src/vs/workbench/contrib/terminal/node/terminalProcess.ts

```diff
@@ -68,20 +68,35 @@ export class TerminalProcess implements ITerminalChildProcess, IDisposable {
 			conptyInheritCursor: true
 		};
 
-		stat(shellLaunchConfig.executable!).then(stat => {
+		const cwdVerification = stat(cwd).catch((err) => {
+			if (err && err.code === 'ENOENT') {
+				// So we can include in the error message the specified CWD
+				shellLaunchConfig.cwd = cwd;
+				return Promise.reject(SHELL_CWD_INVALID_EXIT_CODE);
+			}
+			return;
+		});
+
+		const exectuableVerification = stat(shellLaunchConfig.executable!).then(stat => {
 			if (!stat.isFile() && !stat.isSymbolicLink()) {
-				return this._launchFailed(stat.isDirectory() ? SHELL_PATH_DIRECTORY_EXIT_CODE : SHELL_PATH_INVALID_EXIT_CODE);
+				return Promise.reject(stat.isDirectory() ? SHELL_PATH_DIRECTORY_EXIT_CODE : SHELL_PATH_INVALID_EXIT_CODE);
```

Comment:
This doesn't seem to work anymore:

```
"terminal.integrated.shell.osx": "/bin"
```

Gives:

<img width="433" alt="Screen Shot 2019-07-05 at 11 08 48 AM" src="https://user-images.githubusercontent.com/2193314/60739055-4870ea80-9f15-11e9-96e4-14c8e1a3562d.png">

Reply:
Please test again. It had to due with me using a .catch rather than a second callback for the OnRejected case

Reply:
Works